src/java/htsjdk/samtools/Cigar.java

```diff
@@ -254,15 +259,69 @@ public boolean equals(final Object o) {
 
         final Cigar cigar = (Cigar) o;
 
-        if (cigarElements != null ? !cigarElements.equals(cigar.cigarElements) : cigar.cigarElements != null)
-            return false;
+        return cigarElements.equals(cigar.cigarElements);
+    }
+    
+    /** build a new cigar from a list of cigar operators */
+    public static Cigar fromCigarOperators(final List<CigarOperator> cigarOperators) {
+        if( cigarOperators == null ) throw new IllegalArgumentException("cigarOperators is null");
+        final List<CigarElement> cigarElementList = new ArrayList<>();
+        int i=0;
+        //find adjacent operators and build list of cigar elements
+        while(i < cigarOperators.size()) {
+            final CigarOperator currentOp = cigarOperators.get(i);
+            int j=i+1;
+            while(j < cigarOperators.size() && cigarOperators.get(j).equals(currentOp)) {
+                j++;
+            }
+            cigarElementList.add(new CigarElement(j-i, currentOp));
+            i = j;
+        }
+        return new Cigar(cigarElementList);
+    }
+    
+    /** shortcut to <code>getCigarElements().iterator()</code> */
+    @Override
+    public Iterator<CigarElement> iterator() {
+        return this.getCigarElements().iterator();
+    }
+    
+    /** returns true if the cigar string contains the given operator */
+    public boolean containsOperator(final CigarOperator operator) {
+        for(final CigarElement cigarElement : this.cigarElements) {
```

Comment:
care to java8-ify this function?

cigarElements.stream().anyMatch(el->el.getOperator()==op);